crml/erc20-peg/src/lib.rs

```diff
@@ -233,10 +295,104 @@ decl_module! {
 				Erc20Meta::insert(address, (symbol, decimals));
 			}
 		}
+
+		#[weight = 1_000_000]
+		/// Sets the claim delay for a given AssetId
+		pub fn set_claim_delay(origin, asset_id: AssetId, min_balance: Balance, delay: T::BlockNumber) {
+			ensure_root(origin)?;
+			ClaimDelay::<T>::insert(asset_id, (min_balance, delay));
+			Self::deposit_event(<Event<T>>::ClaimDelaySet(asset_id, min_balance, delay));
+		}
 	}
 }
 
 impl<T: Config> Module<T> {
+	/// Process claims at a block after a delay
+	fn process_claims_at(now: T::BlockNumber, max_claims: u8) -> u8 {
+		let mut removed = 0_u8;
+		for (block, claim_id, pending_claim) in ClaimSchedule::<T>::iter()
+			.filter(|(block, _, _)| block <= &now)
+			.take(max_claims.into())
+		{
+			removed += 1;
+			match pending_claim {
+				PendingClaim::Deposit((deposit_claim, tx_hash)) => {
+					Self::process_deposit_claim(deposit_claim, tx_hash);
+				}
+				PendingClaim::Withdrawal(withdrawal_message) => {
+					// At this stage it is assumed that a mapping between erc20 to asset id exists for this token
+					let asset_id = Self::erc20_to_asset(withdrawal_message.token_address).unwrap_or_default();
```

Comment:
better to fail here than use default asset Id as that has unintended consequences i.e. turns into an Eth withdraw. doesn't seem possible for it to fail but better to be defensive
```suggestion
										let asset_id = Self::erc20_to_asset(withdrawal_message.token_address).unwrap_or_default();
```